### PR TITLE
fix(51920): "Add missing members" quick fixes incorrectly add TypeScript types in JavaScript files

### LINKED ISSUE #51920: "Add missing members" quick fixes incorrectly add TypeScript types in JavaScript files
In a JavaScript file:

```js
// @ts-check

/**
 * @type {{ f: (x: string) => number }}
 */
export const obj = {}
```

Trigger the quick fix on `obj` for "add missing properties"

Currently, the quick fix will create functions that use TypeScript type annotations, even though those are not allowed in JavaScript files.

```js
// @ts-check

/**
 * @type {{ f: (x: string) => number }}
 */
export const obj = {
    f: function (x: string): number {
        throw new Error("Function not implemented.");
    }
}
```

This becomes more apparent with the upcoming decorators implementation as per https://github.com/microsoft/TypeScript/pull/50820#issuecomment-1354052311:

> ```js
> // @ts-check
> 
> export class SomeClass {
>     @someDecorator
>     foo() {
>         return { outputValue: obj.inputValue };
>     }
> }
> ```
> 
> Trigger the "add missing function" quick fix on `someDecorator`. It generates a function with TypeScript type annotations even if it's in a JavaScript file.
> 
> ```js
> // @ts-check
> 
> export class SomeClass {
>     @someDecorator/**/
>     foo() {
>         return { outputValue: obj.inputValue };
>     }
> }
> function someDecorator /**/(target: () => { outputValue: any; }, context: ClassMethodDecoratorContext<SomeClass, () => { outputValue: any; }> & { name: "foo"; private: false; static: false; }): void | (() => { outputValue: any; }) {
>     throw new Error("Function not implemented.");
> }
> ```

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #51920

### CODE REVIEW

**Review by DanielRosenwasser — APPROVED** [review]

File: src/services/codefixes/helpers.ts [hunk 1]
```diff
@@ -386,11 +387,13 @@ export function createSignatureDeclarationFromSignature(
             }
         }
         const newParameters = sameMap(parameters, parameterDecl => {
-            const importableReference = tryGetAutoImportableReferenceFromTypeNode(parameterDecl.type, scriptTarget);
-            let type = parameterDecl.type;
-            if (importableReference) {
-                type = importableReference.typeNode;
-                importSymbols(importAdder, importableReference.symbols);
+            let type = isJs ? undefined : parameterDecl.type;
```
Comment (gabritto) [inline src/services/codefixes/helpers.ts:390]: `parameterDecl.questionToken` below should also be `undefined` if we're in JS.

Comment (a-tarasyuk) [reply src/services/codefixes/helpers.ts:390]: Oke.

Comment (gabritto) [reply src/services/codefixes/helpers.ts:390]: I guess Daniel merged this while I was reviewing, but @a-tarasyuk if you could fix this minor thing, that'd be great. Thanks!

Comment (a-tarasyuk) [reply src/services/codefixes/helpers.ts:390]: I'll fix.

Comment (DanielRosenwasser) [reply src/services/codefixes/helpers.ts:390]: Sorry about that - good catch @gabritto!

Comment (a-tarasyuk) [reply src/services/codefixes/helpers.ts:390]: @gabritto Thanks

**Pull request merged**